### DISCUSSION TITLE
The Parallel transport direction in computing Jacobian

### DISCUSSION THREAD

**Discussion by huweiATgithub** [loco-3d/crocoddyl#1126] [Q&A]
In semi-explicit Euler integration, I am confused about the Jtransport part of `calcDiff`: https://github.com/loco-3d/crocoddyl/blob/0500e398861564b6986d99206a1e0ccec0d66a33/include/crocoddyl/core/integrator/euler.hxx#L76-L115

Let us take $F_u$ as an example. The code is computing $\frac{dx_{n+1}}{du_n} =\frac{\partial x_{n+1}}{\partial \delta x} \frac{\partial \delta x}{\partial u_n} $.
I understand the first part of the code is computing $\frac{\partial \delta x}{\partial u_n} $, and then call `JintegrateTransport` to do the transportation, i.e. left multiply $\frac{\partial x_{n+1}}{\partial \delta x} $:
```c++
state_->JintegrateTransport(x, d->dx, d->Fu, second); 
```

If I understand correctly, $\frac{\partial \delta x}{\partial u_n}$ has column vectors in tangent space at $x_n$. The desired quantity $\frac{dx_{n+1}}{du_n}$ has column vectors in tangent space at $x_{n+1}$.
Therefore, we are transporting vectors from tangent space at $x_n$ to that at $x_{n+1}=x_{n}\oplus\delta x$.

But in the documentation of `JintegrateTransport`, it says: 
> This function performs the parallel transportation of an input matrix whose columns are expressed in the tangent space at $x\oplus\delta x$ to the tangent space at  $x$ point.

It is the reverse of my argument above.
What is wrong with my arguments?

Thanks!

**Comment by cmastalli** [comment]
> But in the documentation of `JintegrateTransport`, it says:
> 
> > This function performs the parallel transportation of an input matrix whose columns are expressed in the tangent space at x⊕δx to the tangent space at  x point.
> 
> It is the reverse of my argument above. What is wrong with my arguments?

Thanks for raising this question. Indeed, the documentation is wrong, it should be the other way around. Could you help us creating a PR that fixes this issue?

**Reply by huweiATgithub** [reply]
Sure, no problem! 
BTW, in the implementation of `JintegrateTransport` of `StateMultibodyTpl`, it calls Pinocchio's `pinocchio::dIntegrateTransport` which says
> this functions transforms a tangent vector expressed at $q\oplus v$ to a tangent vector expressed at $q$

https://github.com/loco-3d/crocoddyl/blob/0500e398861564b6986d99206a1e0ccec0d66a33/include/crocoddyl/multibody/states/multibody.hxx#L211-L229

I don't know whether Pinocchio's documentation is wrong or if I should call Pinocchio reversely.

**Reply by cmastalli** [reply]
Pinocchio documentation seems to be wrong, our code is correct and tested.
